### PR TITLE
Create a cask which represents our base packages

### PR DESCRIPTION
**Pull request by AWare**
Rather than making changes to strap, I was thinking that the best way of specifying our dependencies was to have a Cask that we can install. 

```sh
brew cask install guardian/devtools/gu-base
```

### CODE REVIEW

**Comment by AWare** [comment]
There seems to have been a change in cask making the requirement on a url actually get checked.

File: Casks/gu-base.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on formula:  'guardian/devtools/ssm'
+
+  # gui apps
+  depends_on cask: 'keepingyouawake'
+  depends_on cask: 'google-chrome'
+  depends_on cask: 'iterm2'
+  depends_on cask: 'caskroom/versions/firefoxdeveloperedition'
+  depends_on cask: 'dropbox'
+  depends_on cask: 'visualvm'
+  depends_on cask: 'intellij-idea'
```
Comment (mchv) [inline Casks/gu-base.rb]: I think you made a spelling error here, should be `sublime`

File: Casks/gu-base.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+  # guardian stuff
+  depends_on formula:  'guardian/devtools/ssm'
+
+  # gui apps
+  depends_on cask: 'keepingyouawake'
+  depends_on cask: 'google-chrome'
+  depends_on cask: 'iterm2'
+  depends_on cask: 'caskroom/versions/firefoxdeveloperedition'
+  depends_on cask: 'dropbox'
```
Comment (mchv) [inline Casks/gu-base.rb]: really?

File: Casks/gu-base.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on formula:  'mtr'
+  depends_on formula:  'tree'
+  depends_on formula:  'unrar'
+  depends_on formula:  'watch'
+
+  # dev langs
+  depends_on cask: 'caskroom/versions/java8'
+  depends_on formula:  'scala'
+  depends_on formula:  'sbt'
+  depends_on formula:  'node'
```
Comment (mchv) [inline Casks/gu-base.rb]: `nvm` rather than `node`

Comment (AWare) [reply Casks/gu-base.rb]: nvm will coexist happily with system node, and yarn works best (apparently) when installed with brew on OSX. 

I did forget to install `yarn` though.

File: Casks/gu-base.rb (hunk 2 above)
Comment (AWare) [reply Casks/gu-base.rb]: sorry, that was a mistake.

File: Casks/gu-base.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
+cask 'gu-base' do
```
[the remark this answers is not in the document]
Comment (AWare) [reply Casks/gu-base.rb]: I chose `gu-base` to be explicit about it being somewhat internal only. It's possible to install something by name only if nothing with that name exists in the more official repos. I wouldn't want anyone who's installed ssm to do something like `brew cask install core` and get all our desktop apps.

**Pull request merged**